### PR TITLE
Handle multi-line array

### LINKED ISSUE #126: Multi-line array recurses infinitely
Using 0.2.11:
```kotlin
@Serializable
data class Config(
  val items: List<String>,
)
```
```toml
items = [
  "hey",
  "hi",
]
```
```kotlin
Toml.decodeFromString(Config.serializer(), toml.readText())
```
```
Exception in thread "main" java.lang.StackOverflowError
	at java.base/java.lang.String.startsWith(String.java:2297)
	at kotlin.text.StringsKt__StringsJVMKt.startsWith(StringsJVM.kt:418)
	at kotlin.text.StringsKt__StringsJVMKt.startsWith$default(StringsJVM.kt:416)
	at com.akuleshov7.ktoml.parsers.StringUtilsKt.trimSymbols(StringUtils.kt:151)
	at com.akuleshov7.ktoml.parsers.StringUtilsKt.trimBrackets(StringUtils.kt:83)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parseArray(TomlValue.kt:247)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:237)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
	at com.akuleshov7.ktoml.tree.TomlArray$Companion.parse(TomlValue.kt:239)
```
Works fine when I change it to
```toml
items = ["hey", "hi"]
```

### PR DESCRIPTION
**Pull request by BOOMeranGG**
Fix #126
 
Hello! And thanks for your library! I love Kotlin and I like KTOML format. But I have never done any parsers before. So, I hope you will be patient with me :)

My idea is to merge multiline values into a single line and then process it as usually. If I missed something or this code needs improvement - I am ready.

### CODE REVIEW

**Comment by Ifropc** [comment]
Thank you very much for this PR! This bug is very annoying and prevents us to use this library in our project. (as soon as I tried to parse example toml from our tests I encountered it...)
I know you are busy, but could you please take a look on it @akuleshov7 @NightEule5 ? 0.3.1 with this fix would also be appreciated!

**Review by akuleshov7 — APPROVED** [review]
Hm, looks very easy! thank you so much

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/TomlParser.kt [hunk 1]
```diff
@@ -59,10 +62,24 @@ public value class TomlParser(private val config: TomlInputConfig) {
                 // Parse the inline comment if any
                 val inlineComment = line.trimComment()
 
-                if (line.isTableNode()) {
-                    if (line.isArrayOfTables()) {
+                // append all multiline values to StringBuilder as one line
+                if (multilineValueBuilt.isNotEmpty() || line.isStartOfMultilineValue()) {
```
Comment (akuleshov7) [inline ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/TomlParser.kt:66]: The only question here is: how do we understand the difference between a valid multiline array and the case, when user forgot to add ']' in the end of the statement?

File: ktoml-core/src/commonTest/kotlin/com/akuleshov7/ktoml/decoders/ArrayDecoderTest.kt [hunk 2]
```diff
@@ -95,9 +176,44 @@ class SimpleArrayDecoderTest {
...
+                    null,
+                    3
+                ]
+                """
+        )
+        assertEquals(listOf<Long?>(1, null, 3), testWithMultilineArray.field)
     }
 
+    @Test
+    fun testMultilineArraysWithComments() {
```
Comment (akuleshov7) [inline ktoml-core/src/commonTest/kotlin/com/akuleshov7/ktoml/decoders/ArrayDecoderTest.kt:194]: Can you please add the test, where there is a missing ']' symbol in the end of multiline string. 
I think we should raise an error in this case

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/TomlParser.kt (hunk 1 above)
Comment (BOOMeranGG) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/TomlParser.kt:67]: Thats a good point! Now I detected it

**Comment by BOOMeranGG** [comment]
Wait, I found a little bug. I'm gonna fix it soon

**Comment by akuleshov7** [comment]
pls also don't forget to add yourself to acknowledgements in readme :)

**Comment by akuleshov7** [comment]
> Thank you very much for this PR! This bug is very annoying and prevents us to use this library in our project. (as soon as I tried to parse example toml from our tests I encountered it...) I know you are busy, but could you please take a look on it @akuleshov7 @NightEule5 ? 0.3.1 with this fix would also be appreciated!

It's not yet easy, as soon as we will merge this PR, we will make a release... 😄

**Comment by BOOMeranGG** [comment]
@akuleshov7 Done

**Comment by akuleshov7** [comment]
> @akuleshov7 Done

thank you, you are awesome!

**Pull request merged**